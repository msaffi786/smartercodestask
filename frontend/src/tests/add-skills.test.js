import React from "react";
import {
  render,
  screen,
  cleanup,
  fireEvent,
  waitFor,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import AddSkill from "../components/add-skill.component";
import { BrowserRouter } from "react-router-dom";
import axios from "axios";
axios.defaults.adapter = require("axios/lib/adapters/http");
afterEach(cleanup);
test("renders Welcome", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <AddSkill />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


test("add new skill", async () => {
  const { getByText, getByLabelText, rerender, getByRole } = render(
    <BrowserRouter>
      <AddSkill />
    </BrowserRouter>
  );

  getByLabelText("Title");
  getByLabelText("Description");
  const inputName = getByLabelText("Title");
  fireEvent.change(inputName, { target: { value: "new jest title" } });
  const inputAge = getByLabelText("Description");
  fireEvent.change(inputAge, { target: { value: "new jest description" } });
  const button = screen.getByText("Submit");
  fireEvent.click(button);
  //   await waitFor(() => fireEvent.click(button)
  expect(inputName.innerText).toBe(undefined);
});
