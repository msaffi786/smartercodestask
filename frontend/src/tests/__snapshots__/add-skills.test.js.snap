// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Welcome 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-4"
          >
            <div
              class="card"
            >
              <div
                class="card-body"
              >
                <div>
                  <div
                    class="navbar-nav ml-auto"
                  >
                    <li
                      class="nav-item"
                    >
                      <a
                        class="nav-link"
                        href="/login"
                      >
                        Login
                      </a>
                    </li>
                    <li
                      class="nav-item"
                    >
                      <a
                        class="nav-link"
                        href="/register"
                      >
                        Sign Up
                      </a>
                    </li>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-8"
          >
            <div
              class="submit-form"
            >
              <div>
                <div
                  class="form-group"
                >
                  <label
                    for="title"
                  >
                    Title
                  </label>
                  <input
                    class="form-control"
                    id="title"
                    name="title"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    for="description"
                  >
                    Description
                  </label>
                  <input
                    class="form-control"
                    id="description"
                    name="description"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="star-rating"
                >
                  <button
                    class="off"
                    data-testid="rating"
                    id="rating"
                    type="button"
                  >
                    <span
                      class="star"
                    >
                      ★
                    </span>
                  </button>
                  <button
                    class="off"
                    data-testid="rating"
                    id="rating"
                    type="button"
                  >
                    <span
                      class="star"
                    >
                      ★
                    </span>
                  </button>
                  <button
                    class="off"
                    data-testid="rating"
                    id="rating"
                    type="button"
                  >
                    <span
                      class="star"
                    >
                      ★
                    </span>
                  </button>
                  <button
                    class="off"
                    data-testid="rating"
                    id="rating"
                    type="button"
                  >
                    <span
                      class="star"
                    >
                      ★
                    </span>
                  </button>
                  <button
                    class="off"
                    data-testid="rating"
                    id="rating"
                    type="button"
                  >
                    <span
                      class="star"
                    >
                      ★
                    </span>
                  </button>
                </div>
                <button
                  class="btn btn-success"
                >
                  Submit
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
